ipv8_service.py

```diff
@@ -128,16 +130,23 @@ def __init__(self, configuration, endpoint_override=None, enable_statistics=Fals
         def on_tick(self):
             if self.endpoint.is_open():
                 with self.overlay_lock:
+                    smooth = self.state_machine_lc.interval//len(self.strategies) if self.strategies else 0
+                    ticker = len(self.strategies)
                     for strategy, target_peers in self.strategies:
                         service = strategy.overlay.master_peer.mid
                         peer_count = len(self.network.get_peers_for_service(service))
+                        start_time = time.time()
                         if (target_peers == -1) or (peer_count < target_peers):
                             # We wrap the take_step into a general except as it is prone to programmer error.
                             try:
                                 strategy.take_step(service)
                             except:
                                 logging.error("Exception occurred while trying to walk!\n" +
                                               ''.join(format_exception(*sys.exc_info())))
+                        ticker -= 1 if ticker else 0
+                        sleep_time = smooth - (time.time() - start_time)
+                        if ticker and sleep_time > 0.01:
+                            time.sleep(sleep_time)
```

Comment:
I assume `sleep_time` is very low? Otherwise, it might be a better idea to twistify this logic.